### PR TITLE
Deployment payload size regression tests

### PR DESCRIPTION
**Pull request by korthout**
## Description

Adds a regression test for the maximum deployment payload size currently possible on a 3-partition cluster.

This also cleans up the other tests related to the size of the deployment's payload in an effort to clarify what they truly verify:
- `CreateLargeDeploymentTest.shouldRejectDeployIfResourceIsTooLarge`
  - tests that the incoming request's maxMessageSize is tested
  - is a regression test against #12591
- `DeploymentRejectionTest.shouldRejectDeploymentIfResourceIsTooLarge`
  - verifies that the deployment command is rejected when its resources fit the max message size but still exceed the batch record size (due to follow-up records)
- `CreateDeploymentTest.shouldRejectDeployIfResourceIsTooLarge`
  - verifies that the client's request is rejected when its resources fit the max message size but still exceed the batch record size (due to follow-up records)

> **Note** This pull request explicitly targets `stable/8.2` as all these tests succeed there, but some will not succeed on `main` due to #13233.

## Related issues

relates to #13233 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by korthout** [comment]
I had to force push due to a style error which I fixed by running `mvnd spotless:apply`:

>Caused by: org.apache.maven.plugin.MojoExecutionException: The following files had format violations:
> src/test/java/io/camunda/zeebe/engine/processing/deployment/MultiPartitionDeploymentLifecycleTest.java
>         @@ -194,7 +194,7 @@
>          ················Bpmn.createExecutableProcess("PROCESS")
>          ····················.startEvent()
>          ····················.documentation(
>         -························"x".repeat((int)·(ByteValue.ofMegabytes(4)/2·-·ByteValue.ofKilobytes(2))))
>         +························"x".repeat((int)·(ByteValue.ofMegabytes(4)·/·2·-·ByteValue.ofKilobytes(2))))
>          ····················.done())
>          ············.deploy();

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/CreateDeploymentTest.java [hunk 1]
```diff
@@ -132,11 +137,19 @@ public void shouldRejectDeployIfProcessIsInvalid() {
...
   public void shouldRejectDeployIfResourceIsTooLarge() {
     // when
+    final var modelThatFitsJustWithinMaxMessageSize =
+        Bpmn.createExecutableProcess("PROCESS")
+            .startEvent()
+            .documentation(
+                "x"
+                    .repeat(
+                        (int) (ByteValue.ofMegabytes(MAX_MSG_SIZE_MB) - ByteValue.ofKilobytes(2))))
+            .done();
```
Comment (remcowesterhoud) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/CreateDeploymentTest.java:147]: 🎉 Always nice to see less bpmn files. You scared me with the 12k lines removed though

**Review by remcowesterhoud — APPROVED** [review]
@korthout So if I understand correct this is only to catch regressions. I don't see the fix anywhere?

If it's only about regression it's all good. I like the tests 👍

**Comment by remcowesterhoud** [comment]
image

🤔

**Comment by korthout** [comment]
> @korthout So if I understand correct this is only to catch regressions. I don't see the fix anywhere?

@remcowesterhoud correct. This is only to test the regression and it will fail on main. We'll need to fix it later. Just wanted to get some tests in place first

**Comment by korthout** [comment]
bors merge

**Comment by remcowesterhoud** [comment]
How do you plan on merging this if it will fail on main 😅

**Pull request merged**